modules/core/src/main/java/org/apache/ignite/internal/binary/builder/BinaryBuilderSerializer.java

```diff
@@ -110,6 +111,16 @@ public void writeValue(BinaryWriterExImpl writer, Object val, boolean forceCol,
             return;
         }
 
+        if (val instanceof BinaryEnumObjectImpl) {
+            BinaryEnumObjectImpl obj = (BinaryEnumObjectImpl)val;
```

Comment:
Please check serialization for `GridBinaryMarshaller.UNREGISTERED_TYPE_ID`. There are some relevant code in `org.apache.ignite.internal.binary.builder.BinaryBuilderEnum#writeTo` and `org.apache.ignite.internal.binary.builder.BinaryEnumArrayLazyValue#writeTo`.

Reply:
thanks for noticing. I'll fix it.